### ISSUE TITLE
enum type with numeric keys

### ISSUE THREAD

**Issue by m0x72** [gcanti/tcomb#93]
Hey!
Enum types require that their keys are strings (https://github.com/gcanti/tcomb/blob/master/index.js#L395)
```js
Enums.is = function (x) {
  return Str.is(x) && map.hasOwnProperty(x);
};
```

However, even though all js array keys are essentially strings, the constructor automatically converts keys to strings. The same holds for `hasOwnProperty()`. Thus the following is completely valid js: 
```js
var t = { 3: 'a', 3.2: 'b'};
t.hasOwnProperty(3); // true
t.hasOwnProperty(3.2); //true
```

To reference, the Standard (http://www.ecma-international.org/publications/files/ECMA-ST/Ecma-262.pdf) (Section 15.4, p. 122) states that any key must be converted to a string:
> A property name P (in the form of a String value) is an array index if and only if ToString(ToUint32(P)) is equal to P and ToUint32(P) is not equal to 2^32-1

Now, for floating point numbers, even though the standard requires a `ToUint32` conversion, floating numbers work in any browser and node implementation as outlined above. (http://stackoverflow.com/questions/6649283/how-are-floating-point-array-indices-interpreted).

I'm sure one could just as well specify a `list` of type `number` to achieve a number-`enum` type, but i'd say it may be easier this way. 

Any thoughts on relaxing the `Str.is(x)` requirement on `enum.is()`? :)

**Comment by gcanti** [comment]
> Any thoughts on relaxing the Str.is(x) requirement on enum.is()? :)

Hi! Thanks for your super detailed issue.
It's ok for me. Do you want to send a PR or I go do it?

**Comment by m0x72** [comment]
Yeah sure, will do!

**Comment by gcanti** [comment]
published in v2.0.0-beta
